app/Http/Controllers/GamemodeController.php

```diff
@@ -0,0 +1,161 @@
+            if(in_array("owner", Auth::user()->LpPlayer->groups())){
+                return view('gamemode.create');
+            }
+        }
+        return redirect(route('gamemode.index'));
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        if(isset(Auth::user()->mc_username)){
+            if(in_array("owner", Auth::user()->LpPlayer->groups())){
+                $name = $request->input('name');
+                $port = $request->input('port');
+                $added_to_server = $request->input('added_to_server');
+                $desription = $request->input('description');
+
+                Gamemode::create([
+                    "gamemode" => $name,
+                    "port" => $port,
+                    "added_to_server" => $added_to_server,
+                    "description" => $desription
+                ]);
+                return redirect(route('gamemode.index')); // #TODO: make an success message appear on index page to state gamemode is created!
+            }
+        }
+        return redirect(route('gamemode.index'));
+
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        $gamemode = Gamemode::all()->where('id', '==', $id)->first();
+        if(isset($gamemode)){
+            return view('gamemode.view')->with(["gamemode" => $gamemode]);
+        }else{
+            return redirect(route('gamemode.index')); // #TODO: make an error message appear on index page to state gamemode not found!
+        }
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($id)
+    {
+        if(isset(Auth::user()->mc_username)){
+            if(in_array("owner", Auth::user()->LpPlayer->groups())){
+                $gamemode = Gamemode::all()->where('id', '==', $id)->first();
+
+                if(isset($gamemode)){
+                    return view('gamemode.edit')->with(["gamemode" => $gamemode]);
+                }else{
+                    return redirect(route('gamemode.index')); // #TODO: make an error message appear on index page to state gamemode not found!
+                }
+            }
+        }
+        return redirect(route('gamemode.index'));
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, $id)
+    {
+        if(isset(Auth::user()->mc_username)){
+            if(in_array("owner", Auth::user()->LpPlayer->groups())){
+
+                $gamemode = Gamemode::findOrFail($id);
```

Comment:
Instead using `findOrFail` function, you could just put `Gamemode $gamemode` for the parameter `$id`.

check this guide, [https://mattstauffer.com/blog/implicit-model-binding-in-laravel-5-2/](url)